common/scripts/changelog/collect.mjs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { copyFile, rm, rename } from 'fs/promises';
+import path from 'path';
+import { REPO_ROOT } from '../lib/constants.mjs';
+import { exec, exec_output } from "../lib/exec.mjs";
+import { CHANGE_DIR, CHANGE_DIR_BETA, CHANGE_DIR_STABLE_TEMP } from './constants.mjs';
+import { generateChangelogs } from './changelog.mjs';
+
+async function main() {
+    const args = process.argv;
+    const buildFlavor = args[2];
+    if (buildFlavor !== 'stable' && buildFlavor !== 'beta') {
+        console.error(`Usage: ${args[0]} ${args[1]} <beta|stable>`);
+        throw new Error(`Unknown build flavor ${buildFlavor}`);
+    }
+
+    await ensureCleanWorkingDirectory();
+
+    const changelogs = await listTargetChangelogs(buildFlavor);
+    console.log('Will update the following CHANGELOGs:');
+    changelogs.forEach(({ target }) => {
+        console.log(`  ${target}`);
+    });
+
+    try {
+        if (buildFlavor === 'beta') {
+            await swapInBetaChangeFiles();
+        }
+        await createTemporaryChangelogs(changelogs);
+        await generateChangelogs();
+        if (buildFlavor === 'beta') {
+            await restoreStableChangeFiles();
+        }
+        await commitChangelogs(changelogs);
+    } finally {
+        await restoreWorkingDirectory()
+    }
+}
+
+async function swapInBetaChangeFiles() {
+    await rename(CHANGE_DIR, CHANGE_DIR_STABLE_TEMP);
+    await rename(CHANGE_DIR_BETA, CHANGE_DIR);
+}
+
+async function restoreStableChangeFiles() {
+    await rename(CHANGE_DIR_STABLE_TEMP, CHANGE_DIR);
```

Comment:
Perhaps also rename `change` -> `change-beta` here? Although it'll be empty / non-existent